src/engine/components/CharacterController.cpp

```diff
@@ -87,9 +87,16 @@ namespace lei3d
 	void CharacterController::PhysicsUpdate()
 	{
 		btTransform characterTrans;
-		m_MotionState->getWorldTransform(characterTrans);
+		// if entity pos changed
+		if (m_Entity.m_ResetTransform) {
+			m_Entity.m_ResetTransform = false;
+			characterTrans = m_Entity.getBTTransform();
+			m_RigidBody->setWorldTransform(characterTrans);
```

Comment:
I think if you just change the rigidbody it will also change the motion state. Not entirely sure